### PR TITLE
Allow building `rust-analyzer-proc-macro-srv` as a standalone tool

### PR DESCRIPTION
**Pull request by WaffleLapkin**
This PR allows building `rust-analyzer-proc-macro-srv` as a standalone tool via `x b proc-macro-srv-cli` (I thought that `x b rust-analyzer-proc-macro-srv` should work, but it doesn't for some reason...). Also this PR adds a copy of `rust-analyzer-proc-macro-srv` binary to `build/{triple}/{stage}/libexec/` when building `rust-analyzer-proc-macro-srv`, so that r-a can pick it up.

This is useful to make r-a (and I assume Intellij IDEA) to expand macros when using a custom, build from source toolchain.

r? @jyn514 
[_zulip thread_](https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Frust-analyzer/topic/How.20to.20fix.20.60UnsupportedABI.60.20for.20custom.20toolchains.3F/near/299040175)

### CODE REVIEW

File: src/bootstrap/tool.rs [hunk 1]
```diff
@@ -772,7 +776,19 @@ impl Step for RustAnalyzerProcMacroSrv {
...
-        })
+        })?;
+
+        // Copy `rust-analyzer-proc-macro-srv` to `build/triple/stageN/libexec/`
+        // so that r-a can use it.
+        let libexec_path = builder
+            .out
+            .join(&*builder.config.build.triple)
+            .join(format!("stage{}", self.compiler.stage))
+            .join("libexec");
```
Comment (jyn514) [inline src/bootstrap/tool.rs:787]: This is not correct for stage0 . Please use builder.sysroot(stage) instead, it has a bunch of logic that accounts for download-rustc as well.

**Review by jyn514 — APPROVED** [review]
Thanks! This makes sense to me. Can you also make a PR to the dev guide suggesting that people build this when they make a custom toolchain? It should go near the suggested workflow section, where we already have page about rustup.

File: src/bootstrap/tool.rs (hunk 1 above)
Comment (WaffleLapkin) [reply src/bootstrap/tool.rs:787]: > `builder.sysroot(stage)`

Ah, so that's how you are supposed to do this

**Comment by WaffleLapkin** [comment]
dev guide PR: https://github.com/rust-lang/rustc-dev-guide/pull/1467

**Comment by jyn514** [comment]
@bors r+

Thanks!

**Pull request merged**